ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLAgentExecutor.java

```diff
@@ -132,6 +137,8 @@ public MLAgentExecutor(
         this.encryptor = encryptor;
         this.mcpConnectorIsEnabled = ML_COMMONS_MCP_CONNECTOR_ENABLED.get(clusterService.getSettings());
         clusterService.getClusterSettings().addSettingsUpdateConsumer(ML_COMMONS_MCP_CONNECTOR_ENABLED, it -> mcpConnectorIsEnabled = it);
+        this.agenticSearchIsEnabled = ML_COMMONS_AGENTIC_SEARCH_ENABLED.get(clusterService.getSettings());
```

Comment:
Can we add the settings here: https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/settings/MLFeatureEnabledSetting.java

Reply:
why do we still need this here. Can't we use `isAgenticSearchEnabled` from MLFeatureEnabledSetting class?

Reply:
please check other code. And also if possible (if you have time), move `mcpConnectorIsEnabled` the same way.